### ISSUE TITLE
irodsReServer leaks memory upon reload of ruleset

### ISSUE THREAD

**Issue by tsmeele** [irods/irods#3496]
Environment iRODS 4.1.8 server, Centos7:   
We experience a significant memory usage increase with the iRODS server processes upon each reload of ruleset and the memory usage does not decline afterwards.   
The problem can be reproduced as follows:
```
-bash-4.2$ ps -o "rss,%mem" -p `pidof irodsReServer`
  RSS %MEM
2297640 59.1
-bash-4.2$ touch core.re
-bash-4.2$ ps -o "rss,%mem" -p `pidof irodsReServer`
  RSS %MEM
2674532 68.8
```

**Comment by trel** [comment]
The `irodsReServer` is a long-running process in 4.1 and prior.  This has been changed in 4.2+, it would be interesting to know if it can be reproduced with the latest release code.

I'm not sure we'll be able to provide a fix for this for 4.1 as this may be an architectural limitation.

@kellerb @xu-hao ?

**Comment by kellerb** [comment]
I believe this issue is the same as [this one](https://github.com/irods/irods/issues/3146), which was fixed in 4.1.10.

**Comment by tsmeele** [comment]
I hahdn't found the other issue before but you're right, therefore this issue can be closed.
